src/components/Layout/Header/hooks.ts

```diff
@@ -0,0 +1,75 @@
+import { useMemo, useState } from 'react'
+import { useLocation } from 'react-router'
+
+import { HeaderProps } from 'components/Layout/Header'
+import { HeaderNavLinksInterface } from 'components/Layout/Header/Navigation'
+
+import useWindowSizes from 'hooks/useWindowSizes'
+
+interface UseNavigationProps {
+  navigationArray: HeaderNavLinksInterface[]
+  handleLinkSelect: (key: string) => void
+}
+
+export const useNavigation = (initialState: HeaderProps['navigation'], responsive: boolean): UseNavigationProps => {
+  const [navObject, setNavObject] = useState(initialState)
+  const location = useLocation()
+
+  useMemo(() => {
```

Comment:
Doubt about hooks. My understanding was that, `useMemo` is for when there's a need to memoize a value.
The way it's being used here looks to me more like a `useEffect` pattern instead. Not the case?

Reply:
`useMemo` is fired before render is hit (https://reactjs.org/docs/hooks-reference.html#usememo) do you think it's bad to run it this way? @Velenir

Reply:
changed to useEffect as it wasn't really necessary to make sideeffects during rendering, thanks!